### PR TITLE
add view.indent_markers

### LINKED ISSUE #654: [feature request] add custom indent char by user
i wanted to change the default indent marker char.'
i couldnt find something in the readme or setup.
i found this in the source code:
```lua
local function get_padding_indent_markers(depth, idx, tree, _, markers)
  local padding = ""
  if depth ~= 0 then
    local rdepth = depth/2
    markers[rdepth] = idx ~= #tree.entries
    for i=1,rdepth do
      if idx == #tree.entries and i == rdepth then
        padding = padding..'└ '
      elseif markers[i] then
        padding = padding..'│ ' -- <-----------
      else
        padding = padding..'  '
      end
    end
  end
  return padding
end
```
i guess here the indent char is located

i just want to specifiy this from a global variable or setup function.
please add this feature.

### PR DESCRIPTION
**Pull request by alex-courtis**
closes #654

### CODE REVIEW

File: lua/nvim-tree/renderer/init.lua [hunk 1]
```diff
@@ -284,4 +284,12 @@ function M.render_hl(bufnr)
   end
 end
 
+function M.setup(opts)
+  M.config = {
+    indent_markers = opts.view.indent_markers,
+  }
+
+  require("nvim-tree.renderer.padding").setup(opts)
+end
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/init.lua:293]: I know renderer needs a big refactor... this is just a small step.

File: README.md [hunk 2]
```diff
@@ -136,6 +135,14 @@ require'nvim-tree'.setup { -- BEGIN_DEFAULT_OPTS
     number = false,
     relativenumber = false,
     signcolumn = "yes",
+    indent_markers = {
```
Comment (kyazdani42) [inline README.md:138]: do you think it would make more sens to move the option in a new `renderer.indent_markers` ?

Comment (alex-courtis) [reply README.md:138]: That works. All the renderer config :g can go in there.

I'll update on the weekend.

File: lua/nvim-tree/legacy.lua [hunk 3]
```diff
@@ -169,6 +169,13 @@ local migrations = {
       o.actions.change_dir.global = vim.g.nvim_tree_change_dir_global == 1
     end
   end,
+
+  nvim_tree_indent_markers = function(o)
+    utils.table_create_missing(o, "view.indent_markers")
+    if o.view.indent_markers.enable == nil then
+      o.view.indent_markers.enable = vim.g.nvim_tree_indent_markers == 1
+    end
+  end,
```
Comment (kyazdani42) [inline lua/nvim-tree/legacy.lua:178]: looks like this hasn't been updated :)

**Review by kyazdani42 — CHANGES_REQUESTED** [review]

File: lua/nvim-tree/legacy.lua (hunk 3 above)
Comment (alex-courtis) [reply lua/nvim-tree/legacy.lua]: Good catch

**Review by kyazdani42 — APPROVED** [review]

**Pull request merged**